chia/full_node/full_node_api.py

```diff
@@ -1331,18 +1317,17 @@ async def request_header_blocks(self, request: wallet_protocol.RequestHeaderBloc
         if request.end_height < request.start_height or request.end_height - request.start_height > 32:
             return None
 
-        header_hashes = []
+        header_hashes: List[bytes32] = []
         for i in range(request.start_height, request.end_height + 1):
             if not self.full_node.blockchain.contains_height(uint32(i)):
                 reject = RejectHeaderBlocks(request.start_height, request.end_height)
                 msg = make_msg(ProtocolMessageTypes.reject_header_blocks, reject)
                 return msg
-            header_hashes.append(self.full_node.blockchain.height_to_hash(uint32(i)))
+            header_hash: Optional[bytes32] = self.full_node.blockchain.height_to_hash(uint32(i))
+            assert header_hash is not None
```

Comment:
I believe that previously this would have just returned no results instead of raising.

Reply:
As noted above, may as well just get the hash and check `None`ness instead of pre-checking and post-asserting.

Reply:
To be explored separately.